core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java

```diff
@@ -40,6 +40,7 @@
 public abstract class CloudObjectInputSource<T extends InputEntity> extends AbstractInputSource
     implements SplittableInputSource<CloudObjectLocation>
 {
+  protected static final int MAX_LISTING_LENGTH = 1024;
```

Comment:
I don't think it's good to have the base class decide the max length for all the implementations. It would be better to expose this through a function. Should this be made configurable? 

I see the `GoogleCloudStorageInputSource` used to have a max of 10. This makes it 1024. Also the AzureInputSource uses it's own definition for `MAX_LISTING_LENGTH`

Reply:
I talked to @clintropolis  about the Google limit and that was a mistake, was meant to be 1024 as well. I figured since they are all 1024 now, we could abstract that to base class, and then override in the specific implementation if necessary, which right now it doesn't seem to be so. I agree that this should be configurable.